src/packagedcode/go_mod.py

```diff
@@ -73,140 +109,221 @@
 github.com/davecgh/go-spew v1.1.1
 
 For example:
->>> ob = GoMod()
-
->>> p = ob.parse_require('require github.com/davecgh/go-spew v1.1.1')
+>>> p = parse_require('require github.com/davecgh/go-spew v1.1.1')
 >>> assert p.group('namespace') == ('github.com/davecgh')
 >>> assert p.group('name') == ('go-spew')
 >>> assert p.group('version') == ('v1.1.1')
 
->>> p = ob.parse_exclude('exclude github.com/davecgh/go-spew v1.1.1')
+>>> p = parse_exclude('exclude github.com/davecgh/go-spew v1.1.1')
 >>> assert p.group('namespace') == ('github.com/davecgh')
 >>> assert p.group('name') == ('go-spew')
 >>> assert p.group('version') == ('v1.1.1')
 
->>> p = ob.parse_dep_link('github.com/davecgh/go-spew v1.1.1')
+>>> p = parse_dep_link('github.com/davecgh/go-spew v1.1.1')
 >>> assert p.group('namespace') == ('github.com/davecgh')
 >>> assert p.group('name') == ('go-spew')
 >>> assert p.group('version') == ('v1.1.1')
 """
 
+# Regex expressions to parse different types of go.mod file dependency
+parse_module = re.compile(
+    r'^module\s'
+    r'(?P<module>[a-z].*)'
+).match
+
+parse_module_name = re.compile(
+    r'^module(\s)*'
+    r'(?P<namespace>(.*))'
+    r'/'
+    r'(?P<name>[^\s]*)'
+).match
+
+parse_require = re.compile(
+    r'^require(\s)*'
+    r'(?P<namespace>(.*))'
+    r'/'
+    r'(?P<name>[^\s]*)'
+    r'\s'
+    r'(?P<version>(.*))'
+).match
+
+parse_exclude = re.compile(
+    r'^exclude(\s)*'
+    r'(?P<namespace>(.*))'
+    r'/'
+    r'(?P<name>[^\s]*)'
+    r'\s'
+    r'(?P<version>(.*))'
+).match
+
+parse_dep_link = re.compile(
+    r'(?P<namespace>(.*))'
+    r'/'
+    r'(?P<name>[^\s]*)'
+    r'\s'
+    r'(?P<version>(.*))'
+).match
+
+
+def preprocess(line):
+    """
+    Return line string after removing commented portion and excess spaces.
+    """
+    if "//" in line:
+        line = line[:line.index('//')]
+    line = line.strip()
+    return line
+
+
+def parse_gomod(location):
+    """
+    Return a dictionary containing all the important go.mod file data.
+    """
+    with io.open(location, encoding='utf-8', closefd=True) as data:
+        lines = data.readlines()
+
+    gomods = {}
```

Comment:
Why not use an object instead of a dict?

Reply:
I was keeping some of the previous format. Now I have updated my PR again. :)